canvas/services/event.py

```diff
@@ -52,6 +55,20 @@ def get_question_stats(question):
                 answers[answer] = 0
             answers[answer] += 1
 
+    most_recent_submission_time_for_submission_author = (
+        submissions.filter(uqj__user=OuterRef("uqj__user")).order_by("-submission_time").values("submission_time")[:1]
+    )
+
+    recent_submissions = (
+        submissions.filter(uqj__user__role="Student")
+        .annotate(recent_timestamp=Subquery(most_recent_submission_time_for_submission_author))
+        .filter(submission_time=F("recent_timestamp"))
+    )
```

Comment:
It is not guaranteed that the last submission is actually the one we consider the final submission. In most cases, we get the maximum of their grades. So if you submit a wrong code after a correct one, we consider the question to be solved correctly.